src/lightning/pytorch/plugins/precision/double.py

```diff
@@ -12,28 +12,76 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from contextlib import contextmanager
-from typing import Any, cast, Generator, List, Literal, Tuple
+from typing import Any, Generator, Literal
 
 import torch
 import torch.nn as nn
 from lightning_utilities.core.apply_func import apply_to_collection
 from torch import Tensor
-from torch.optim import Optimizer
 
 import lightning.pytorch as pl
 from lightning.fabric.plugins.precision.utils import _convert_fp_tensor
-from lightning.pytorch.overrides.base import _LightningPrecisionModuleWrapperBase
+from lightning.fabric.utilities.device_dtype_mixin import _DeviceDtypeModuleMixin
 from lightning.pytorch.plugins.precision.precision_plugin import PrecisionPlugin
+from lightning.pytorch.utilities.rank_zero import rank_zero_deprecation
 
 
-class LightningDoublePrecisionModule(_LightningPrecisionModuleWrapperBase):
+class DoublePrecisionPlugin(PrecisionPlugin):
```

Comment:
Reviewer: Sorry for the confusing diff. I moved the `LightningDoublePrecisionModule` to the bottom of the file, and this shows up as deleting and adding the precision plugin at the top.